src/manifests/test_manifest.py

```diff
@@ -117,12 +130,21 @@ def __create__(self, data: dict) -> 'TestComponent':
         return TestComponent(data)
 
 
+class TestComponentTopology:
+    def __init__(self, data: dict):
+        if data is not None:
+            self.cluster_configs = data
+        else:
+            self.cluster_configs = [{'cluster_name': 'cluster1', 'data_nodes': 1, 'manager_nodes': 0}]  # type: ignore[assignment]
```

Comment:
Continue with the class hierarchy just like for `TestComponentTopology`. Add a `ClusterConfig` class that has properties such as `cluster_name` and `data_nodes`, so that the caller doesn't have to deal with JSON.